sklearn/utils/tests/test_estimator_checks.py

```diff
@@ -727,7 +730,9 @@ def test_all_estimators_all_public():
 def test_xfail_ignored_in_check_estimator():
     # Make sure checks marked as xfail are just ignored and not run by
     # check_estimator(), but still raise a warning.
-    assert_warns(SkipTestWarning, check_estimator, NuSVC())
+    with warnings.catch_warnings(record=True) as records:
+        check_estimator(NuSVC())
+    assert SkipTestWarning in [rec.category for rec in records]
```

Comment:
`with pytest.warns` does not work in this case? Why? It's special cased in pytest?